commands.go

```diff
@@ -1188,6 +1190,18 @@ func (c cmdable) LPop(ctx context.Context, key string) *StringCmd {
 	return cmd
 }
 
+func (c cmdable) LPos(ctx context.Context, key string, value string, rank int64, maxlen int64) *IntCmd {
+	cmd := NewIntCmd(ctx, "lpos", key, value, "rank", rank, "maxlen", maxlen)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+func (c cmdable) LPosCount(ctx context.Context, key string, value string, rank int64, count int64, maxlen int64) *IntSliceCmd {
+	cmd := NewIntSliceCmd(ctx, "lpos", key, value, "rank", rank, "count", count, "maxlen", maxlen)
+	_ = c(ctx, cmd)
+	return cmd
+}
```

Comment:
What if we replace these 2 commands with one:

```go
type LPosArgs struct {
    Rank int64
    Count int64
    MaxLen int64
}

LPos(ctx context.Context, key, value string, args LPosArgs)
```

Reply:
To get you more context in case you missed it:
LPos with Count **NOT** set returns an integer or nil if not found.
LPos with Count set returns an array, empty array if none found.

As you suggested we can have only one command that returns an array. I do prefer having only one command than two.

Both Count and MaxLen default to 0 in Redis, but Rank's default is 1. Setting Rank to 0 results in a failure. 

We could be helpful and set good default ourselves.

From a user perspective and in the simplest case possible (get the first pos of an item) which of the following options would you prefer?
```go
// 1 -  We'll check that Rank is 0 and set it to 1
client.Lpos(context.Background(), "list", "item", LPosArgs{}) 

// 2 - We'll check for nil and do the defaults ourselves, Rank=1, Count=0, MaxLen = 0
client.Lpos(context.Background(), "list", "item", nil)

// 3 - User needs to always set rank to 1, just LPosArgs{} would fail.
client.Lpos(context.Background(), "list", "item", LPosArgs{ Rank: 1 }) 
```
I prefer option 2. We set good defaults and if the user wants more control they'll need to provide a struct with correct settings.

What are your thoughts?

Reply:
Thanks for the details! What about

```go
type LPosArgs struct {
    Rank int64
    MaxLen int64
}

LPos(ctx context.Context, key, value string, args LPosArgs) *IntCmd
LPosCount(ctx context.Context, key, value string, count int64, args LPosArgs) *IntSliceCmd
```

It is slightly more readable than just using positional arguments.